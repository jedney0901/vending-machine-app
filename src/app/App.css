.App {
    text-align: center;
}

.App-logo {
    animation: App-logo-spin infinite 20s linear;
    height: 80px;
}

.App-header {
    background-color: #222;
    height: 150px;
    padding: 20px;
    color: white;
}

.App-intro {
    font-size: large;
}

@keyframes App-logo-spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}

.bordered {
    border-style: solid;
    border-width: 3px;
    border-color: black;
}
