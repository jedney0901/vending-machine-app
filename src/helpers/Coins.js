var Coins = [
  {
    amount: 10,
    type: '$0.1'
  },
  {
    amount: 20,
    type: '$0.2'
  },
  {
    amount: 50,
    type: '$0.5'
  },
  {
    amount: 100,
    type: '$1'
  },
  {
    amount: 200,
    type: '$2'
  }
]

export default Coins
