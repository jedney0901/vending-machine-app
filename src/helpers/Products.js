var Products = [
  {
    name: 'Caramel',
    cost: 250,
  },
  {
    name: 'Hazelnut',
    cost: 310,
  },
  {
    name: 'Organic Raw',
    cost: 200,
  }
]

export default Products
