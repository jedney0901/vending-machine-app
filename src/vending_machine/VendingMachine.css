.vendingMachine {
    height: 400px;
    max-width: 30%;
    margin: 5%;
}