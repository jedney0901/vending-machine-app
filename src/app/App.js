import React, {Component} from 'react';
import './App.css';
import VendingMachine from '../vending_machine/vendingMachine'

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentOrder: {
        credit: 0,
        selected: 'Please select what vegan bar you would like',
        spent: 0
      }
    }
  };

  render() {
    return (
      <div className="App">
        Hello World
        <VendingMachine />
      </div>
    );
  }
}

export default App;
