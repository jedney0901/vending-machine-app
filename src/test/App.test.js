import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import App from '../app/App.js';
import renderer from 'react-test-renderer';
import VendingMachine from '../vending_machine/VendingMachine';

// helpers
import Products from '../helpers/Products'
import Coins from '../helpers/Coins'

describe('App Component', () => {
  const mockAddCredit = jest.fn();
  const mockSelectProducts = jest.fn();
  const mockResetOrder = jest.fn();
  const mockConfirmOrder = jest.fn();

  let initialState = {
    credit: 0,
    selected: [],
    confirm: 'no'
  }

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  describe('render App component', () => {
    const app = shallow(<App />);

    it('should add update the credit amount', () => {
      const coin = 50
      const expected = initialState.credit += coin;
      const actual = app.updateCredit(coin);

      expect(actual).toEqual(expected);
    });

    it('should reset the order', () => {
      const expected = {
        selected: [],
        credit: 0
      };

      const actual = app.instance().resetOrder();
      expect(actual).toEql(expected);
    });

    it('should select products for their purchase', () => {
      const credit = 0.5
      const expected = alert('There is not enough credit!')
      const actual = app.instance().selectProducts(credit);

      expect(actual).toEqual(expected);
    });
  });

  describe('VendingMachine Component', () => {
    const component = shallow(
      <VendingMachine />
    );

    it('renders and matches our snapshot', () => {
      const component = renderer.create(
        <VendingMachine
          userCredit={initialState.credit}
          CurrentOrder={initialState.selected}
          ConfirmationStatus={initialState.confirm}
          addCredit={mockAddCredit}
          addProduct={mockSelectProducts}
          resetOrder={mockResetOrder}
          confirmOrder={mockConfirmOrder}
          coins={Coins}
          products={Products}
        />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});