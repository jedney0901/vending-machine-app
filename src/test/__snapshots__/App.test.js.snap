// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component VendingMachine Component renders and matches our snapshot 1`] = `
<div
  className="vendingMachine"
  style={
    Object {
      "color": "#70C5CA",
    }
  }
>
  Hi there, welcome to your healthy vending machine, enter your coins in to proceed.
  <div>
    <button
      className=""
      data-react-toolbox="button"
      disabled={false}
      href={undefined}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "#262626",
          "borderRadius": "10px",
          "color": "white",
          "height": "40px",
          "margin": "20px",
          "width": "60px",
        }
      }
      type="button"
    >
       
      $0.1
       
    </button>
    <button
      className=""
      data-react-toolbox="button"
      disabled={false}
      href={undefined}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "#262626",
          "borderRadius": "10px",
          "color": "white",
          "height": "40px",
          "margin": "20px",
          "width": "60px",
        }
      }
      type="button"
    >
       
      $0.2
       
    </button>
    <button
      className=""
      data-react-toolbox="button"
      disabled={false}
      href={undefined}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "#262626",
          "borderRadius": "10px",
          "color": "white",
          "height": "40px",
          "margin": "20px",
          "width": "60px",
        }
      }
      type="button"
    >
       
      $0.5
       
    </button>
    <button
      className=""
      data-react-toolbox="button"
      disabled={false}
      href={undefined}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "#262626",
          "borderRadius": "10px",
          "color": "white",
          "height": "40px",
          "margin": "20px",
          "width": "60px",
        }
      }
      type="button"
    >
       
      $1
       
    </button>
    <button
      className=""
      data-react-toolbox="button"
      disabled={false}
      href={undefined}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "#262626",
          "borderRadius": "10px",
          "color": "white",
          "height": "40px",
          "margin": "20px",
          "width": "60px",
        }
      }
      type="button"
    >
       
      $2
       
    </button>
  </div>
  Your current credit is $
  0.5
  , please choose your product's once you have enough credit.
  <div>
    <button
      className=""
      data-react-toolbox="button"
      disabled={false}
      href={undefined}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "#F3C719",
          "borderRadius": "10px",
          "height": "40px",
          "margin": "20px",
          "width": "60px",
        }
      }
      type="button"
    >
       
      Caramel
       $
      2.5
       
    </button>
    <button
      className=""
      data-react-toolbox="button"
      disabled={false}
      href={undefined}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "#F3C719",
          "borderRadius": "10px",
          "height": "40px",
          "margin": "20px",
          "width": "60px",
        }
      }
      type="button"
    >
       
      Hazelnut
       $
      3.1
       
    </button>
    <button
      className=""
      data-react-toolbox="button"
      disabled={false}
      href={undefined}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "#F3C719",
          "borderRadius": "10px",
          "height": "40px",
          "margin": "20px",
          "width": "60px",
        }
      }
      type="button"
    >
       
      Organic Raw
       $
      2
       
    </button>
  </div>
  <button
    className=" confirmOrder"
    data-react-toolbox="button"
    disabled={false}
    href={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "color": "white",
        "margin": "20px",
      }
    }
    type="button"
  >
    Purchase
  </button>
  <button
    className=" resetOrder"
    data-react-toolbox="button"
    disabled={false}
    href={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "color": "white",
        "margin": "20px",
      }
    }
    type="button"
  >
    Reset
  </button>
  <div>
    <div />
  </div>
</div>
`;
